tools/backups2datalad.py

```diff
@@ -736,5 +761,65 @@ def pdb_excepthook(exc_type, exc_value, tb):
         pdb.post_mortem(tb)
 
 
+def mkrelease(repo: Path, dandiset: RemoteDandiset, commitish: str = None) -> None:
+    # `dandiset` must have its version set to the published version
+    if commitish is None:
+        remote_assets = list(dandiset.get_assets())
+        repo_assets = json.loads(
+            (repo / ".dandi" / "assets.json").read_text(encoding="utf-8")
+        )
+        if assets_eq(remote_assets, repo_assets):
+            commitish = "HEAD"
+        else:
+            for cmt in subprocess.check_output(
+                ["git", "rev-list", "--skip=1", "HEAD", "--", ".dandi/assets.json"],
+                cwd=repo,
+                universal_newlines=True,
+            ).splitlines():
+                repo_assets = json.loads(
+                    subprocess.check_output(
+                        ["git", "show", f"{cmt}:.dandi/assets.json"],
+                        cwd=repo,
+                        universal_newlines=True,
+                    )
+                )
+                if (
+                    repo_assets
+                    and not isinstance(repo_assets[0], dict)
+                    or "asset_id" not in repo_assets[0]
+                ):
+                    # We've reached an older, pre-dandi-api version of
+                    # assets.json; we're not going to find the assets we're
+                    # after here
+                    break
+                if assets_eq(remote_assets, repo_assets):
+                    commitish = cmt
+                    break
+            if commitish is None:
+                raise RuntimeError(
```

Comment:
I was thinking about this case:  if no commit found, the script should 
- for the `committish` take the latest commit before the release (modified) date

and then below we should not just tag the commitish, we should

- also compare the assets (if were not compared before for that committish) or doesn't hurt to always do 
- and if listing differs: create a "release commit" (could be some throw away branch `release-{dandiset.version_id}` on top of the `commitish` where we populate from the list of assets which is known to that remote versioned dandiset
- and only then tag that new "release commit"

this way we can fully automate under a reasonable assumption of when (in git history) release happened, and ensure that we are tagging exactly the same state of assets as the released dandiset

Reply:
@yarikoptic 

> where we populate from the list of assets

Do you mean essentially running `update-from-backup` for that release?

Reply:
yes!

Reply:
note that it should not `datalad save -d^` that commit in the super dataset -- super dataset (this repo) should just keep tracking `master` branches corresponding to drafts. (i wish I thought about it before -- we should have made default branch to be `draft` -- would have aligned nicely with what we have on the server)